src/Resolver/Resolver.php

```diff
@@ -149,7 +154,18 @@ public function loadOrCreateAndPrepareElement(array $inputData, bool $createNew
             $this->getCreateLocationStrategy()->updateParent($element, $inputData);
             $justCreated = true;
         } else {
+            $oldParentId = $element->getParentId();
             $this->getLocationUpdateStrategy()->updateParent($element, $inputData);
+
+            // The parent of a variant cannot be changed anymore.
+            if (
```

Comment:
I'm just wondering, if we could pack also that check into the FindParentStrategy? 
Then only the FindParentStrategy needs to know about the variant handling...

Reply:
@turbo-ele :ping_pong:  any thoughts on this?

Reply:
@fashxp sorry, somehow I missed the first message.
I would say moving it into the strategy is not an option, as a variant - correct me if I'm wrong - should never be moved from it's parent. So if we'd move it into the FindParentStrategy, another location update strategy, such as the StaticPathStrategy, would otherwise be able to move the variant into a folder.
Alternatively, this check has to be implemented in every strategy individually, which i wanted to avoid.

Reply:
you are completely right!